src/tests.rs

```diff
@@ -261,7 +261,7 @@ fn test_region_tag() -> Result<(), Box<dyn std::error::Error>> {
 
     let data = match region_entry.data {
         IndexData::Bin(d) => d,
-        _ => return Err(Box::new(RPMError::new("should be bin"))),
+        _ => return Err(Box::new(RPMError::Other("should be bin".to_string()))),
```

Comment:
Avoiding generic errors would be a big plus, every variant should have one root cause. A `other` is the uneeded escape hatch here. I'd rather have this be named specifically to this use-case.